app/packages/core/src/components/Sidebar/ViewSelection/index.tsx

```diff
@@ -82,107 +72,107 @@ export default function ViewSelection(props: Props) {
 
   const items =
     (data as { savedViews: [fos.State.SavedView] })?.savedViews || [];
-  const isEmptyView = !loadedView?.length;
-
-  const viewOptions: DatasetViewOption[] = [
-    DEFAULT_SELECTED,
-    ...items.map((item: fos.State.SavedView) => {
-      const { name, urlName, color, description, viewStages } = item;
 
-      return {
+  const viewOptions: DatasetViewOption[] = useMemo(
+    () => [
+      DEFAULT_SELECTED,
+      ...map(items, ({ name, color, description, urlName, viewStages }) => ({
         id: urlName,
         label: name,
-        color: color,
-        description: description || "",
+        color,
+        description,
+        slug: urlName,
         viewStages,
-      };
-    }),
-  ] as DatasetViewOption[];
-
-  // TODO: get saved views here from state and pass as items instead
-  const [viewSearch, setViewSearch] = useRecoilState<string>(viewSearchTerm);
+      })),
+    ],
+    [items]
+  );
 
-  const searchedData: DatasetViewOption[] = filter(
-    viewOptions,
-    ({ id, label, description }: DatasetViewOption) => {
-      return (
-        id === "1" ||
-        label?.toLowerCase().includes(viewSearch) ||
-        description?.toLowerCase().includes(viewSearch)
-      );
-    }
-  ) as DatasetViewOption[];
+  const searchData: DatasetViewOption[] = useMemo(
+    () =>
+      filter(
+        viewOptions,
+        ({ id, label, description, slug }: DatasetViewOption) =>
+          id === "1" ||
+          label.toLowerCase().includes(viewSearch) ||
+          description?.toLowerCase().includes(viewSearch) ||
+          slug?.toLowerCase().includes(viewSearch)
+      ) as DatasetViewOption[],
+    [viewOptions, viewSearch]
+  );
 
-  useEffect(() => {
-    if (isEmptyView) {
-      setSelected(viewOptions[0]);
-      setSavedViewParam(null);
-    }
-  }, [loadedView]);
+  const loadedView = useRecoilValue<fos.State.Stage[]>(fos.view);
```

Comment:
I think relying on state here may cause issues (like the page flashing I mentioned) if the session and relay get out of sync. We may need to refactor this section to use graphQL rather than states.

Do you think this is a concern @benjaminpkane?

Reply:
good point on multiple tabs being open and some flashing if you switch between views. I'll work on fixing that today. any recommendation is welcomed of course!

Reply:
note that this code is already in your rebased branch - this branch a cleaner version